version: '3.9'
services:
#  TODO: currently I am encountering connection refused to Postgres. Please proceed running the app from your local
#  rest:
#    container_name: golang_container
#    tty: true
#    environment:
#      - POSTGRES_USER=${POSTGRES_USER}
#      - POSTGRES_PASSWORD=${POSTGRES_PASSWORD}
#      - POSTGRES_DB=${POSTGRES_DB}
#      - POSTGRES_HOST=${POSTGRES_HOST}
#    build:
#      context: .
#      dockerfile: cmd/rest/Dockerfile
#    ports:
#      - 8080:8080
#    restart: on-failure
#    volumes:
#      - .:/app
#    depends_on:
#      - postgresdb
#    networks:
#      - cookbook

  postgresdb:
    image: postgres:latest
    container_name: postgres_container
    environment:
      - POSTGRES_USER=${POSTGRES_USER}
      - POSTGRES_PASSWORD=${POSTGRES_PASSWORD}
      - POSTGRES_DB=${POSTGRES_DB}
      - POSTGRES_HOST=${POSTGRES_HOST}
    ports:
      - '${POSTGRES_PORT}:5432'
    volumes:
      - ./dev/postgresql_data:/var/lib/postgresql/data
    networks:
      - cookbook

volumes:
  postgresql_data:

networks:
  cookbook:
    driver: bridge